crates/flux-middle/src/rty/mod.rs

```diff
@@ -1099,6 +1133,7 @@ impl BaseTy {
             | BaseTy::Array(_, _)
             | BaseTy::Closure(_, _)
             | BaseTy::Coroutine(..)
+            | BaseTy::Dynamic(_, _)
```

Comment:
There's something I don't quite understand yet. We are putting `Dynamic` in `BaseTy` which means they can be indexed (by `unit` as returned by this function), however, the concrete type under the dynamic type could be indexed by another sort. I don't know if these should be somehow related or if they are completely orthogonal.

Reply:
I think its fine, because once its `Dynamic` you have *zero* information about the underlying concrete type, everything is abstracted to whatever is know about the `dyn`...